framework/src/base/FEProblem.C

```diff
@@ -1252,7 +1250,7 @@ FEProblem::addKernel(const std::string & kernel_name, const std::string & name,
   parameters.set<FEProblem *>("_fe_problem") = this;
   if (_displaced_problem != NULL && parameters.get<bool>("use_displaced_mesh"))
   {
-    parameters.set<SubProblem *>("_subproblem") = _displaced_problem;
+    parameters.set<SubProblem *>("_subproblem") = _displaced_problem.get();
```

Comment:
For the record... this is why I don't like SharedPointer stuff... because you always ultimately end up just pulling it back out as a regular pointer anyway.

I'm not against it here... and I'm not lobbying for you to change it back... just standing on my soap box :wink: